compiler/rustc_infer/src/infer/error_reporting/nice_region_error/different_lifetimes.rs

```diff
@@ -145,84 +146,85 @@ impl<'a, 'tcx> NiceRegionError<'a, 'tcx> {
             }
         }
 
-        self.suggest_adding_lifetime_params(sub, ty_sup, ty_sub, &mut err);
+        if suggest_adding_lifetime_params(self.tcx(), sub, ty_sup, ty_sub, &mut err) {
+            err.note("each elided lifetime in input position becomes a distinct lifetime");
+        }
```

Comment:
Note was originally in `suggest_adding_lifetime_params` but by moving it here we can avoid having the note under NLL.
However if we do want the note, I can just move it back to where it comes from

Reply:
Hmm. The note is kind of redundant with the NLL error output, I guess.